src/node_dependencies/index.js

```diff
@@ -43,36 +52,48 @@ const isNotJunk = function (file) {
 }
 
 // Retrieve all the files recursively required by a Node.js file
-const getDependencies = async function (mainFile, srcDir, pluginsModulesPath) {
+const getDependencies = async function (mainFile, srcDir, pluginsModulesPath, featureFlags) {
   const packageJson = await getPackageJson(srcDir)
   const state = getNewCache()
 
   try {
-    return await getFileDependencies({ path: mainFile, packageJson, state, pluginsModulesPath })
+    return await getFileDependencies({ featureFlags, path: mainFile, packageJson, pluginsModulesPath, state })
   } catch (error) {
     error.message = `In file "${mainFile}"\n${error.message}`
     throw error
   }
 }
 
-const getFileDependencies = async function ({ path, packageJson, pluginsModulesPath, state, treeShakeNext }) {
+const getFileDependencies = async function ({
+  featureFlags,
+  path,
+  packageJson,
+  pluginsModulesPath,
+  state,
+  treeShakeNext,
+}) {
   if (state.localFiles.has(path)) {
     return []
   }
 
   state.localFiles.add(path)
 
   const basedir = dirname(path)
-  // This parses JavaScript in `path` to retrieve all the `require()` statements
-  // TODO: `precinct.paperwork()` uses `fs.readFileSync()` under the hood,
-  // but should use `fs.readFile()` instead
-  const dependencies = precinct.paperwork(path, { includeCore: false })
+  const dependencies = featureFlags.parseWithEsbuild
```

Comment:
We'll use this feature flag to roll this out progressively. Once the rollout is complete, we can remove `precinct` entirely.